exp/prepare_sr_experiment.py

```diff
@@ -0,0 +1,51 @@
+import os
+import sys
+import pickle
+
+from data.data_loading import load_dataset, load_graph_dataset
+from data.perm_utils import permute_graph, generate_permutation_matrices
+from definitions import ROOT_DIR
+
+__families__ = [
+    'sr16622',
+    'sr251256',
+    'sr261034',
+    'sr281264',
+    'sr291467',
+    'sr351668',
+    'sr351899',
+    'sr361446',
+    'sr401224'
+]
+
+def prepare(family, jobs, max_ring_size, permute, init, seed):
```

Comment:
Yes, this is to generate the datasets in advance, caching them to disk. This can take a significant amount of time in certain families, and thought it was convenient to have a separate routine for it. It can also support the generation of permuted version of the SR dataset that are then used in the tests. I can rename as `preprocess_sr_graphs`?

Reply:
Oh I see. So this is not just for the permuted graphs?

Reply:
It can be used for both, but I think in practice it is only called by the unit tests

Reply:
Then should we call it `prepare_sr_test` to make it explicit this is for testing only? We probably want to separate it more clearly from the code of the actual experiment.